dvc/repo/diff.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import unicode_literals
+
+import os
+
+
+from dvc.scm.base import FileNotInCommitError
+import dvc.scm.base as git
+
+
+def _extract_dir(self, output):
+    """Extract the content of dvc tree file
+    Args:
+        self(object) - Repo class instance
+        output(object) - OutputLOCAL class instance
+    Returns:
+        dict - dictionary with keys - paths to file in .dvc/cache
+                               values -checksums for that files
+    """
+    lst = output.dir_cache
+    return {i["relpath"]: i["md5"] for i in lst}
+
+
+def _get_tree_changes(self, a_entries, b_entries):
+    result = {
+        git.DIFF_DEL: 0,
+        git.DIFF_IDENT: 0,
+        git.DIFF_CHANGE: 0,
+        git.DIFF_NEW: 0,
+        git.DIFF_MOVE: 0,
+    }
+    keys = set(a_entries.keys())
+    keys.update(b_entries.keys())
+    diff_size = 0
+    for key in keys:
+        if key in a_entries and key in b_entries:
+            if a_entries[key] == b_entries[key]:
+                result[git.DIFF_IDENT] += 1
+            else:
+                result[git.DIFF_CHANGE] += 1
+                diff_size += os.path.getsize(
+                    self.cache.local.get(b_entries[key])
+                ) - os.path.getsize(self.cache.local.get(a_entries[key]))
+        elif key in a_entries:
+            result[git.DIFF_DEL] += 1
+            diff_size -= os.path.getsize(self.cache.local.get(a_entries[key]))
+        else:
+            result[git.DIFF_NEW] += 1
+            diff_size += os.path.getsize(self.cache.local.get(b_entries[key]))
+    result[git.DIFF_SIZE] = diff_size
+    return result
+
+
+def _get_diff_outs(self, diff_dct):
+    self.tree = diff_dct[git.DIFF_A_TREE]
+    a_outs = {str(out): out for st in self.stages() for out in st.outs}
+    self.tree = diff_dct[git.DIFF_B_TREE]
+    b_outs = {str(out): out for st in self.stages() for out in st.outs}
+    outs_paths = set(a_outs.keys())
+    outs_paths.update(b_outs.keys())
+    results = {}
+    for path in outs_paths:
+        if path in a_outs and path in b_outs:
+            results[path] = {
+                "a_output": a_outs[path],
+                "b_output": b_outs[path],
+                git.DIFF_NEW_FILE: False,
+                "deleted_file": False,
```

Comment:
Why DIFF_NEW_FILE is present, but "deleted_file" is left as a string?

Reply:
These keys are local for that file, I decided to keep them as they are. Don't you agree with that?

Reply:
yes, but another point is also to not copy-paste, as that might be error-prone, especially when working with dicts, since one typo might lead to bug.

Reply:
fixed that